addon-test-support/-private/filter-test-modules.js

```diff
@@ -1,3 +1,38 @@
+const MODULE_PATH_REGEXP = /^(!?)\/([\w\W]*)\/(i?$)/;
+
+/**
+ * Return the matched test.
+ * e.g. if an input is '!/weight/' it returns an array, ['!/weight/', '!', 'weight', ''];
+ *
+ * @param {*} modulePathFilter
+ */
+function getRegexFilter(modulePathFilter) {
+  return MODULE_PATH_REGEXP.exec( modulePathFilter );
+}
+
+/**
+ * Return a list of test modules that contain a given module path string.
+ *
+ * @param {*} modules
+ * @param {*} modulePathFilter
+ */
+function stringFilter(modules, modulePathFilter) {
+  return modules.filter( module => module.includes(modulePathFilter) );
+}
+
+/**
+ * Return a list of test modules that matches with a given regular expression.
+ *
+ * @param {*} tests
+ * @param {*} modules
+ * @param {*} modulePathRegexFilter
```

Comment:
what does this param look like? is it possible for it to be undefined or null? cause if so, line 31 and 32 would throw an error